### ISSUE TITLE
ROMLoader-like function for MSU-1 support

### ISSUE THREAD

**Issue by codemann8** [zig-for/snfm#13]
It'd be nice if this utility gives similar function to that of RomLoader, being able to easily move sfc roms to sd2snes (for MSU-1 usage) with minimal user interaction. For a user point of view, when RomLoader is configured, all a user needs to do is double click on an sfc file, a command prompt will appear with a menu listing all the MSU-1 packs that the user configured. The user enters their choice (picking a number) and hits enter. The ROM gets transferred to a certain directory and the resulting filename is specific per the user config. And after it is done transferring, the ROM auto boots up.

Quick primer on how MSU-1 works:
An MSU pack consists of many PCM files, they all exist in the same folder, all files use the same prefix with a `-` and track number following it. ie. `alttp_msu-7.pcm`. In order for this MSU pack to function, the ROM using it must also have the same file prefix. ie. `alttp_msu.sfc`. Also, most people have several MSU packs, so typically people will keep them separated by way of holding them in separate folders (altho you could technically have one mega folder with all the PCMs, as long as they all have different prefixes).

So, a user will have already established a mass of MSU packs on their SD cards, so you won't have to add any additional support to any large file transferring. In fact, this whole thing doesn't even need to be aware of any of these files existing. All it would need to do is recognize the incoming file matches a certain pattern, prompt the user for a choice of destination (aka. which MSU pack), transfer the file appropriately, run the ROM

tldr list:
- Allows users to set up and SFC file association to snfc, for easy double-click on sfc file support
- Pattern matches the source filename, determining what destination and filename it will write the file to
- Allowing the user to pick from multiple options (if user config has multiple options ie. MSU-packs)
- Transfers the ROM to the correct destination
- Auto boots the ROM

Let me know if you need any other details. Also, I can probably help with a PR or two, altho my C/++ knowledge is extremely lacking so I'd prob need a lot of help.

Below is an example yaml file used as user configuration in ROMLoader:

```yaml
### device option can be commented out to have this tool use the first sd2snes device it finds, which in most cases is fine.
### If you have multiple sd2snes units connected, you'll need to specify the one you want to use.

# device: "COM3"

### default directory to put the ROM if no rules are matched.
### Make sure this directory exists on your SD card (will try to create it if /romloader)
default_destination: "/ROMs"

### a set of rules to use to put certain ROMs in certain locations, such as your randomizer ROMs, useful for MSU1 users
rules:
  alttpr:
    name_pattern:
      - "ALttP - VT_*" # look for an input ROM that matches this name
      - "alttpr - *"
      - "alttprcustom - *"
      - "ER_*"
      - "DR_*"
      - "OR_*"
      - "*DR_*"
      - "BD_*"
      - "Daily Challenge_*"
    destinations:
      - name: default
        path: "/ROMs/_alttpr"
      - name: WildArms2
        path: "/ROMs/_alttpr/_msu1/wa2"
        romname: alttp_msu.sfc
      - name: FinalFantasy7
        path: "/ROMs/_alttpr/_msu1/ff7"
        romname: alttp_msu.sfc
      - name: FinalFantasy7Remake
        path: "/ROMs/_alttpr/_msu1/ff7r"
        romname: alttp_msu.sfc
      - name: Naruto
        path: "/ROMs/_alttpr/_msu1/Naruto"
        romname: alttp_msu.sfc
      - name: Kirby and the Forgotten Land
        path: "/ROMs/_alttpr/_msu1/KirbyForgottenLand"
        romname: alttp_msu.sfc
  smz3:
    name_pattern:
      - "SMALttP - sm-*"
    destinations:
      - name: default
        path: "/ROMs/_smz3"
  smw:
    name_pattern:
      - "smw-*"
    destinations:
      - name: default
        path: "/ROMs/SuperMarioWorld/Rando"
  supermetroiditem:
    name_pattern:
      - "Item Randomizer *"
      - "VARIA_Randomizer_*"
    destinations:
      - name: default
        path: "/ROMs/SuperMetroid/Rando"```
